File: Data Structure/Array or Matrix/Medium/LET_no1010_Pairs of Songs With Total Durations Divisible by 60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        record = collections.defaultdict(int)
        res = 0
        for cur in time:
            cur %= 60
            res += record[(60 - cur) % 60]
            record[cur] += 1
            
        return res

File: Data Structure/Array or Matrix/Medium/LET_no1010_Pairs of Songs With Total Durations Divisible by 60.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        int res = 0;
        vector<int> modTable(60, 0);
        for (int t : time) {
            res += modTable[(60 - t % 60) % 60];
            modTable[t % 60]++;
        }
        return res;
    }
};